### PR TITLE
Add a backwards compatible change for enabling lower casing

### PR DESCRIPTION
**Pull request by fjy**

### CODE REVIEW

File: src/main/java/com/metamx/common/parsers/JSONParser.java [hunk 1]
```diff
@@ -68,23 +68,33 @@ public Object apply(JsonNode node)
...
+    this.parserConfig = parserConfig;
+  }
+
   @Deprecated
   public JSONParser(Iterable<String> fieldNames)
   {
     this(new ObjectMapper(), fieldNames, null);
   }
 
+  @Deprecated
```
Comment (fjy) [inline src/main/java/com/metamx/common/parsers/JSONParser.java]: i think we shouldn't have fieldNames and excludes at this level

Comment (fjy) [inline src/main/java/com/metamx/common/parsers/JSONParser.java]: I actually think we should rework the constructor to be
Objectmapper, fieldNames, jsonparseconfig

I don't think excludes should be at this level, or if they are, be in the jsonparseconfig

File: src/main/java/com/metamx/common/parsers/JSONToLowerParser.java [hunk 2]
```diff
@@ -0,0 +1,82 @@
+package com.metamx.common.parsers;
```
Comment (xvrl) [inline src/main/java/com/metamx/common/parsers/JSONToLowerParser.java]: license

File: src/main/java/com/metamx/common/parsers/JSONToLowerParser.java [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+    try {
+      Map<String, Object> map = new LinkedHashMap<>();
+      JsonNode root = objectMapper.readTree(input);
+
+      Iterator<String> keysIter = (fieldNames == null ? root.fieldNames() : fieldNames.iterator());
+
+      while (keysIter.hasNext()) {
+        String key = keysIter.next().toLowerCase(); // only difference from JSONParser parse()
+
+        if (exclude.contains(key)) {
```
Comment (xvrl) [inline src/main/java/com/metamx/common/parsers/JSONToLowerParser.java]: should exclude be lowercased too?

**Pull request merged**